Change error count messages

citizen428 opened the pull request:
See #33525 for details. r? @Mark-Simulacrum

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @Mark-Simulacrum (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 2aa69f3 has been approved by `Mark-Simulacrum`

<!-- @bors r=Mark-Simulacrum 2aa69f34ab0640b9649cce914f91b516cf003045 -->

bors commented:
:hourglass: Testing commit 2aa69f3 with merge a90475c...

bors commented:
:broken_heart: Test failed - [status-travis](https://travis-ci.org/rust-lang/rust/builds/234794110?utm_source=github_status&utm_medium=notification)

Mark-Simulacrum commented on src/librustc_errors/lib.rs:
```diff
@@ -501,9 +501,8 @@ impl Handler {
-            1 => s = "aborting due to previous error".to_string(),
             _ => {
-                s = format!("aborting due to {} previous errors", self.err_count.get());
+                s = format!("aborting due to previous error(s)".to_string);
```
The call to `to_string` should just be removed (or made into a call and the `format!` removed.

arielb1 commented:
Looks like you need to update the UI tests:
```
[01:04:45] ---- [ui] ui/suggestions/confuse-field-and-method/private-field.rs stdout ----
[01:04:45] 	ui: /checkout/src/test/ui/suggestions/confuse-field-and-method/private-field.rs
[01:04:45] normalized stderr:
[01:04:45] error: no method named `dog_age` found for type `animal::Dog` in the current scope
[01:04:45]   --> $DIR/private-field.rs:26:23
[01:04:45]    |
[01:04:45] 26 |     let dog_age = dog.dog_age();
[01:04:45]    |                       ^^^^^^^ private field, not a method
[01:04:45] 
[01:04:45] error: aborting due to previous error(s)
[01:04:45] 
[01:04:45] 
[01:04:45] 
[01:04:45] expected stderr:
[01:04:45] error: no method named `dog_age` found for type `animal::Dog` in the current scope
[01:04:45]   --> $DIR/private-field.rs:26:23
[01:04:45]    |
[01:04:45] 26 |     let dog_age = dog.dog_age();
[01:04:45]    |                       ^^^^^^^ private field, not a method
[01:04:45] 
[01:04:45] error: aborting due to previous error
[01:04:45] 
[01:04:45] 
[01:04:45] 
[01:04:45] diff of stderr:
[01:04:45] 
[01:04:45]  error: no method named `dog_age` found for type `animal::Dog` in the current scope
[01:04:45]    --> $DIR/private-field.rs:26:23
[01:04:45]     |
[01:04:45]  26 |     let dog_age = dog.dog_age();
[01:04:45]     |                       ^^^^^^^ private field, not a method
[01:04:45]  
[01:04:45] -error: aborting due to previous error
[01:04:45] +error: aborting due to previous error(s)
```

There's some command that updates the tests automatically, but I don't remember what it is. @Mark-Simulacrum do you remember how to run it?

citizen428 commented:
@arielb I was wondering if there's a command, the ones I already updated I used `sed` for. Clearly I missed some, will look into it.

Mark-Simulacrum commented:
Wasn't able to find or get someone to answer me about the potential script, but looks like Travis failed with on `ui/codemap_tests/two_files.rs`.

citizen428 commented:
@Mark-Simulacrum Yes, I fixed that already, that's the build that's currently running (I had a previous commit where I fixed almost all of the others, but this one slipped through). Fingers crossed this will finally be green. I just wish tests would run faster on my local machine, I feel bad always tying uo Travis.

citizen428 commented:
Ok, this is getting ridiculous, I had amended and force pushed an old checkout (switched from home to work computer).... I'll need to take some time to fix this again, sorry for all the noise 🤦‍♂️

citizen428 commented:
@bors retry

Mark-Simulacrum commented:
No problem! Just to check, in case you weren't aware, `./x.py test -i --stage 1 src/test/ui` is probably the best way to run them locally today.

@bors r+

bors commented:
:pushpin: Commit 5558c64 has been approved by `Mark-Simulacrum`

<!-- @bors r=Mark-Simulacrum 5558c64f33446225739c1153b43d2e309bb4f50e -->
